files/common/var/lib/delphix-platform/ansible/10-delphix-platform/roles/delphix-platform/tasks/main.yml

```diff
@@ -368,3 +368,26 @@
   with_items:
     - login
     - sshd
+
+#
+# The Delphix platform supports running as an NFSv4-only server, which
+# means that the services used by NFSv3 can be disabled (masked).
+# Unfortunately, the nfs-server unit contains a Requires dependency on
+# the nfs-mountd.service, preventing the nfs-mountd.service from being
+# masked without also preventing the nfs-server from starting. We
+# change this dependency to a Wants dependency so that if the service is
+# enabled, it will be available for NFSv3 configurations, but it can be
+# disabled for NFSv4 configurations.
+#
+- copy:
+    remote_src: yes
+    src: /lib/systemd/system/nfs-server.service
+    dest: /etc/systemd/system/nfs-server.service
+    owner: root
+    group: root
+    mode: 0644
+
+- replace:
+    path: /etc/systemd/system/nfs-server.service
+    regexp: '^Requires(=nfs-mountd.service)$'
```

Comment:
To future-proof this, should we also add a check after this statement that the replacement was actually performed? If `nfs-server.service` is modified by upstream and another service is added to the `Requires=nfs-mountd.service` line, this replacement will fail.

Reply:
Good idea, I'll look at how to do that.

Reply:
I would suggest we either do something like:
```
regexp: '^Requires='
replace: 'Wants='
```

Or use linux-pkg and modify the file in a downstream repository, if we're worried about upstream modifications misbehaving with this replacement. IMO, failing the platform service is the wrong place to notify us of upstream making incompatible changes to the service definition (which is what a check would do).